pyproject.toml

```diff
@@ -130,14 +130,14 @@ packages = ["src/jupytext", "src/jupytext_config", "jupyterlab/jupyterlab_jupyte
 "jupyterlab/jupyter-config" = "etc/jupyter"
 "jupyterlab/jupyterlab_jupytext/labextension" = "share/jupyter/labextensions/jupyterlab-jupytext"
 
-[tool.hatch.build.hooks.jupyter-builder]
+[tool.hatch.build.hooks.custom]
 # Enable running hook by default.
 # We disable this hook only by setting env var HATCH_BUILD_HOOKS_ENABLE=false
 # So `HATCH_BUILD_HOOKS_ENABLE=false pip install .` will **not** build JupyterLab related
 # extension. A simple `pip install .` will **always** build extension
 enable-by-default = true
```

Comment:
Why not set `enable-by-default = false`, so that extension will not build by default. In the CI, we will need to add `HATCH_BUILD_HOOKS_ENABLE=true` to force the building of extension.

Reply:
Well sure that would solve this issue indeed! That's how we were doing it up to version 1.14.x I believe. Would that be ok with you? I'll be careful to make sure that we don't miss the extension in the conda package too.

Reply:
From my understanding, most of the `jupytext` "contributors" are interested in developing core `jupytext` and do not touch its extension. If that is the case, I think it makes more sense to disable extension building by default as it completely avoids `node` dependency. 

So, yes, I think it makes sense for me to set `enable-by-default = false` and more over, that is what I proposed originally when we moved to `hatch` build system.